src/server/routes/entity/entity.js

```diff
@@ -519,20 +519,30 @@ module.exports.createEntity = (
 				return model.forge(propsToSet)
 					.save(null, {method: 'insert', transacting});
 			})
-			.then((entityModel) =>
-				achievement.processEdit(req.user.id)
-					.then(() => entityModel)
-			)
 			.then(
 				(entityModel) =>
 					entityModel.refresh({
 						withRelated: ['defaultAlias'],
 						transacting
 					})
 			)
-			.then((entity) => entity.toJSON());
+			.then((entity) =>
+				({
+					entityJSON: entity.toJSON(),
+					editorJSON
+				})
+
+			);
 	});
 
+	entityCreationPromise.then((creationJSON) =>
+		achievement.processEdit(
+			creationJSON.editorJSON.id,
+			creationJSON.entityJSON.revisionId
+		)
+			.then(() => creationJSON.entityJSON)
+	);
+
 	return handler.sendPromiseResult(
 		res,
 		entityCreationPromise,
```

Comment:
The processing is fine, but just assign this new promise to a variable, then send that to the handler, so that the route doesn't respond to the client before it's finished processing:

    const achievementPromise = entityCreationPromise.then((creationJSON) => ...

    return handler.sendPromiseResult(
        res,
        achievementPromise,